File: Tugas Besar/Matakuliah.php
<?php
//Simpanlah dengan nama file : Matakuliah.php
require_once 'database.php';
class Matakuliah 
{
    private $db;
    private $table = 'matakuliah';
    public $kode_matakuliah = "";
    public $nama_matakuliah = "";
    public $sks = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_kode_matakuliah(int $kode_matakuliah)
    {
        $query = "SELECT * FROM $this->table WHERE kode_matakuliah = $kode_matakuliah";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`kode_matakuliah`,`nama_matakuliah`,`sks`) VALUES ('$this->kode_matakuliah','$this->nama_matakuliah','$this->sks')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET kode_matakuliah = '$this->kode_matakuliah', nama_matakuliah = '$this->nama_matakuliah', sks = '$this->sks' 
        WHERE id_matakuliah = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_kode_matakuliah($kode_matakuliah): int
    {
        $query = "UPDATE $this->table SET kode_matakuliah = '$this->kode_matakuliah', nama_matakuliah = '$this->nama_matakuliah', sks = '$this->sks' 
        WHERE kode_matakuliah = $kode_matakuliah";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE id_matakuliah = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_kode_matakuliah($kode_matakuliah): int
    {
        $query = "DELETE FROM $this->table WHERE kode_matakuliah = $kode_matakuliah";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>

File: Tugas Besar/Matakuliah.py
import requests
import json
class Matakuliah:
    def __init__(self):
        self.__id=None
        self.__kode_matakuliah = None
        self.__nama_matakuliah = None
        self.__sks = None
        self.__url = "http://a0832601.xsph.ru/AppTugasBesar/matakuliah_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def kode_matakuliah(self):
        return self.__kode_matakuliah
        
    @kode_matakuliah.setter
    def kode_matakuliah(self, value):
        self.__kode_matakuliah = value
    @property
    def nama_matakuliah(self):
        return self.__nama_matakuliah
        
    @nama_matakuliah.setter
    def nama_matakuliah(self, value):
        self.__nama_matakuliah = value
    @property
    def sks(self):
        return self.__sks
        
    @sks.setter
    def sks(self, value):
        self.__sks = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_kode_matakuliah(self, kode_matakuliah):
        url = self.__url+"?kode_matakuliah="+kode_matakuliah
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['id_matakuliah']
            self.__kode_matakuliah = item['kode_matakuliah']
            self.__nama_matakuliah = item['nama_matakuliah']
            self.__sks = item['sks']
        return data
    def simpan(self):
        payload = {
            "kode_matakuliah":self.__kode_matakuliah,
            "nama_matakuliah":self.__nama_matakuliah,
            "sks":self.__sks
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_kode_matakuliah(self, kode_matakuliah):
        url = self.__url+"?kode_matakuliah="+kode_matakuliah
        payload = {
            "kode_matakuliah":self.__kode_matakuliah,
            "nama_matakuliah":self.__nama_matakuliah,
            "sks":self.__sks
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_kode_matakuliah(self,kode_matakuliah):
        url = self.__url+"?kode_matakuliah="+kode_matakuliah
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text